bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/handler/RingDeviceHandler.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ring.handler;
+
+import static org.openhab.binding.ring.RingBindingConstants.*;
+
+import java.math.BigDecimal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.ring.internal.RingDeviceRegistry;
+import org.openhab.binding.ring.internal.data.RingDevice;
+import org.openhab.binding.ring.internal.errors.DeviceNotFoundException;
+import org.openhab.binding.ring.internal.errors.IllegalDeviceClassException;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+
+/**
+ * The {@link RingDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Wim Vissers - Initial contribution
+ * @author Ben Rosenblum - Updated for OH4 / New Maintainer
+ */
+
+@NonNullByDefault
+public abstract class RingDeviceHandler extends AbstractRingHandler {
+
+    /**
+     * The RingDevice instance linked to this thing.
+     */
+    protected @NonNullByDefault({}) RingDevice device;
+
+    public RingDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Link the device, and update the device with the status CONFIGURED.
+     *
+     * @param id the device id
+     * @param deviceClass the expected class
+     * @throws DeviceNotFoundException when device is not found in the RingDeviceRegistry.
+     * @throws IllegalDeviceClassException when the registered device is of the wrong type.
+     */
+    protected void linkDevice(String id, Class<?> deviceClass)
+            throws DeviceNotFoundException, IllegalDeviceClassException {
+        device = RingDeviceRegistry.getInstance().getRingDevice(id);
+        if (device.getClass().equals(deviceClass)) {
```

Comment:
`deviceClass.equals(device.getClass())`, to ensure null-safety

Reply:
fixed.